tests/TestCase.php

```diff
@@ -10,7 +10,65 @@
 namespace Slim\Tests;
 
 use PHPUnit\Framework\TestCase as PhpUnitTestCase;
+use Prophecy\Argument;
+use Prophecy\Prophecy\ObjectProphecy;
+use ReflectionProperty;
+use Slim\Views\Twig;
+use Slim\Views\TwigExtension;
+use Slim\Views\TwigMiddleware;
+use Slim\Views\TwigRuntimeExtension;
+use Slim\Views\TwigRuntimeLoader;
 
 abstract class TestCase extends PhpUnitTestCase
 {
+    /**
+     * Create a twig prophecy given a uri prophecy and a base path.
+     *
+     * @param ObjectProphecy $uriProphecy
+     * @param string         $basePath
+     *
+     * @return ObjectProphecy
+     */
+    protected function createTwigProphecy(ObjectProphecy $uriProphecy, string $basePath): ObjectProphecy
+    {
+        $self = $this;
+
+        $twigProphecy = $this->prophesize(Twig::class);
+
+        /** @noinspection PhpUndefinedMethodInspection */
+        $twigProphecy
+            ->addExtension(Argument::type('object'))
+            ->will(function ($args) use ($self) {
+                /** @var TwigExtension $extension */
+                $extension = $args[0];
+
+                $self->assertEquals('slim', $extension->getName());
+            })
+            ->shouldBeCalledOnce();
+
+        /** @noinspection PhpUndefinedMethodInspection */
+        $twigProphecy->
+        addRuntimeLoader(Argument::type('object'))
+            ->will(function ($args) use ($self, $uriProphecy, $basePath) {
+                /** @var TwigRuntimeLoader $runtimeLoader */
+                $runtimeLoader = $args[0];
+                $runtimeExtension = $runtimeLoader->load(TwigRuntimeExtension::class);
+
+                $self->assertInstanceOf(TwigRuntimeExtension::class, $runtimeExtension);
+
+                /** @var TwigRuntimeExtension $runtimeExtension */
+                $self->assertSame($uriProphecy->reveal(), $runtimeExtension->getUri());
+                $self->assertSame($basePath, $runtimeExtension->getBasePath());
+            })
+            ->shouldBeCalledOnce();
+
+        return $twigProphecy;
+    }
+
+    protected function assertInaccessiblePropertySame($expected, object $o, string $name)
```

Comment:
Please remove the `object` type-hinting.

Reply:
I'll do it but why?

Reply:
Done

Reply:
Because `object` is valid from PHP 7.2.0 on only. But this project is PHP 7.1 compliant.

Reply:
Okay